resource_booking/models/resource_booking.py

```diff
@@ -13,6 +13,38 @@
 from odoo.addons.resource.models.resource import Intervals
 
 
+def _availability_is_fitting(available_intervals, start_dt, end_dt):
+    start_date = start_dt.date()
+    end_date = end_dt.date()
+    # Booking is uninterrupted on the same calendar day.
+    if len(available_intervals) == 1 and available_intervals._items[0][:2] == (
+        start_dt,
+        end_dt,
+    ):
+        return True
+    # Booking spans more than one calendar day, e.g. from 23:00 to 1:00
+    # the next day.
+    elif (
+        available_intervals
+        and start_date != end_date
+        and (end_date - start_date).days == (len(available_intervals) - 1)
+    ):
```

Comment:
We could improve this check by removing it, and iterating over every item to check whether they are on subsequent dates, keeping a rolling tally.

Reply:
wrong PR